src/platform_impl/linux/mod.rs

```diff
@@ -626,7 +626,7 @@ unsafe extern "C" fn x_error_callback(
     display: *mut x11::ffi::Display,
     event: *mut x11::ffi::XErrorEvent,
 ) -> c_int {
-    let xconn_lock = X11_BACKEND.lock().unwrap();
+    let xconn_lock = X11_BACKEND.lock().unwrap_or_else(|e| e.into_inner());
```

Comment:
I usually prefer:
```suggestion
    let xconn_lock = X11_BACKEND.lock().unwrap_or_else(PoisonError::into_inner);
```
Purely a preference thing.

Reply:
I don't like the verbosity here.